forest/src/cli/wallet_cmd.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+#![allow(unused_variables, unused_imports)]
+
+use super::stringify_rpc_err;
+use rpc_client::{
+    balance, default, export, import, list, new, new_client, set_default, sign, verify,
+};
+use structopt::StructOpt;
+use wallet::{get_sig_type, KeyInfo};
+
+#[derive(Debug, StructOpt)]
+pub enum WalletCommands {
+    /// Prints out wallet balance of specified address via RPC
+    #[structopt(about = "Get account balance")]
+    Balance {
+        #[structopt(short, help = "Input a valid address")]
+        address: String,
+    },
+
+    /// Creates wallet of specified signature type via RPC
+    #[structopt(
+        about = "Generate a new key of the given type",
+        help = "Generate a new key of the given type"
+    )]
+    New {
+        #[structopt(
+            short,
+            default_value = "secp256k1",
+            help = "Input signature type [bls | secp256k1] (default secp256k1"
+        )]
+        sig_type: String,
+    },
+
+    /// Prints out all addresses in the wallet via RPC
+    #[structopt(about = "List wallet address", help = "List wallet address")]
+    List,
+
+    /// Marks the given address as as the default wallet via RPC
+    #[structopt(
+        about = "Set default wallet address",
+        help = "Set default wallet address"
+    )]
+    SetDefault {
+        #[structopt(short, help = "Input valid address")]
+        address: String,
+    },
+
+    /// Prints out the address marked as default in the wallet via RPC
+    #[structopt(
+        name = "default",
+        about = "Get default wallet address",
+        help = "Get default wallet address"
+    )]
+    Def,
+
+    /// Prints out private key of specified address in wallet via RPC
+    #[structopt(about = "Exports wallet keys", help = "Exports wallet keys")]
+    Export {
+        #[structopt(short, help = "Input valid address")]
+        address: String,
+    },
+
+    /// Signs the given bytes using the given address via RPC
+    #[structopt(about = "Sign a message", help = "Sign a message")]
+    Sign {
+        #[structopt(short, help = "Must specify signing address")]
+        address: String,
+        #[structopt(short, help = "Must specify message to sign")]
+        message: String,
+    },
+}
+
+impl WalletCommands {
+    pub async fn run(&self) {
+        // TODO handle cli config
```

Comment:
I don't really understand this TODO, I can guess that it's referring to specifying the url to send requests?